src/UI/mainwindow.h

```diff
@@ -0,0 +1,21 @@
+#ifndef MAINWINDOW_H
+#define MAINWINDOW_H
+
+#include <QMainWindow>
+
+QT_BEGIN_NAMESPACE
+namespace Ui { class MainWindow; }
+QT_END_NAMESPACE
+
+class MainWindow : public QMainWindow
+{
+    Q_OBJECT
+
+public:
+    MainWindow(QWidget *parent = nullptr);
+    ~MainWindow();
+
+private:
+    Ui::MainWindow *ui;
```

Comment:
Please place it on stack (remove *) and delete destructor. I reccomend you to name class members with prefix m_: `m_ui`

Reply:
renamed ui to m_ui and placed it on stack
deleted destructor